src/weights.jl

```diff
@@ -682,6 +682,35 @@ function mean(A::AbstractArray, w::UnitWeights; dims::Union{Colon,Int}=:)
     return mean(A, dims=dims)
 end
 
+"""
+    mean(f, A::AbstractArray, w::AbstractWeights)
+
+Compute the weighted mean of array `A`, after transforming it'S
+contents with the function `f`, with weight vector `w` (of type
+`AbstractWeights`).
+
+# Examples
+```julia
+n = 20
+x = rand(n)
+w = rand(n)
+mean(√, x, weights(w))
+```
+"""
+mean(f, A::AbstractArray, w::AbstractWeights) =
+_funcweightedmean(f, A, w)
```

Comment:
I don't remember, was there any particular reason for introducing `_funcweightedmean` instead of implementing two methods for `mean`?

Reply:
No there wasn't any such reason. I am removing this function, and implementing it as a method for `mean`